plugins/modules/iosxr_user.py

```diff
@@ -278,66 +269,30 @@ def copy_key_to_node(self, base64keyfile):
         src = base64keyfile
         dst = "/harddisk:/publickey_%s.b64" % (name)
 
-        ssh = paramiko.SSHClient()
-        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        if not ssh_keyfile:
-            ssh.connect(node, username=user, password=password)
-        else:
-            ssh.connect(node, username=user, allow_agent=True)
-        sftp = ssh.open_sftp()
-        sftp.put(src, dst)
-        sftp.close()
-        ssh.close()
+        copy_file(self._module, src, dst)
 
     def addremovekey(self, command):
         """ Add or remove key based on command
         """
-        provider = self._module.params.get("provider") or {}
-        node = provider.get("host")
-        if node is None:
-            return False
-
-        user = provider.get("username")
-        if user is None:
-            return False
-
-        password = provider.get("password")
-        ssh_keyfile = provider.get("ssh_keyfile")
-
-        ssh = paramiko.SSHClient()
-        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        if not ssh_keyfile:
-            ssh.connect(node, username=user, password=password)
-        else:
-            ssh.connect(node, username=user, allow_agent=True)
-        ssh_stdin, ssh_stdout, ssh_stderr = ssh.exec_command(
-            "%s \r" % (command)
-        )
-        readmsg = ssh_stdout.read(
-            100
-        )  # We need to read a bit to actually apply for some reason
-        if (
-            ("already" in readmsg)
-            or ("removed" in readmsg)
-            or ("really" in readmsg)
-        ):
-            ssh_stdin.write("yes\r")
-        ssh_stdout.read(
-            1
-        )  # We need to read a bit to actually apply for some reason
-        ssh.close()
-
-        return readmsg
+
+        conn = get_connection(self._module)
+        out = conn.send_command(command, prompt="yes/no", answer="yes")
+
+        return out
 
     def run(self):
+        warning_message = (
+            "Operation with key file failed, couldn't determine host or user to connect. "
+            "Please setup provider argument or pass its arguments directly to module in "
+            "case of using local connection plugin before running this playbook"
```

Comment:
```suggestion
            "If using local connection, please ensure provider argument is set "
            "before running this playbook."
```
I _think_ this is closer to what you mean to say? Again, module arguments have been removed, so there's no direct argument passing.

Reply:
With last fixes this message seems to lost its sense at all

For my point of view there is no situation where it can be used, and this message and all related code can be deleted

Reply:
If it is no longer relevant, then please do remove it, and all the places where it is used as well.